lints/lint_idn_dnsname_must_be_nfkc.go

```diff
@@ -0,0 +1,51 @@
+package lints
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+	"golang.org/x/net/idna"
+	"golang.org/x/text/unicode/norm"
+	"strings"
+)
+
+type IDNNotNFKC struct {
+	// Internal data here
+}
+
+func (l *IDNNotNFKC) Initialize() error {
+	return nil
+}
+
+func (l *IDNNotNFKC) CheckApplies(c *x509.Certificate) bool {
+	return util.IsExtInCert(c, util.SubjectAlternateNameOID)
+}
+
+func (l *IDNNotNFKC) RunTest(c *x509.Certificate) (ResultStruct, error) {
+	for _, dns := range c.DNSNames {
+		splitLabels := strings.Split(dns, ".")
+		for _, label := range splitLabels {
+			if strings.HasPrefix(label, "xn--") {
```

Comment:
Are you sure this is the correct way of catching punycode? We should probably match whatever regexp is used here: https://github.com/letsencrypt/boulder/commit/1209102e7ee23eeb9ed39cc2c0b6c1f62e05310a#diff-e613fb1bb146bc2843103b461bef02c8R217.

Reply:
Yes, this is correct.

Reply:
👍